src/main/java/org/opensearch/neuralsearch/plugin/NeuralSearch.java

```diff
@@ -80,6 +80,7 @@ public Collection<Object> createComponents(
         final IndexNameExpressionResolver indexNameExpressionResolver,
         final Supplier<RepositoriesService> repositoriesServiceSupplier
     ) {
+        NeuralSearchClusterUtil.instance().initialize(clusterService);
```

Comment:
We should start getting rid of these kind of inits and move towards singleton pattern without this kind of inits. 

May be an AI for maintainers